src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/HoistedThisTests.cs

```diff
@@ -950,11 +949,22 @@ static int P
 
         private void VerifyHasThis(string source, string methodName, string expectedType, string expectedIL)
         {
-            var comp = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugDll, assemblyName: ExpressionCompilerUtilities.GenerateUniqueName());
-            var runtime = CreateRuntimeInstance(comp);
+            var sourceCompilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugDll, assemblyName: ExpressionCompilerUtilities.GenerateUniqueName());
+            var runtime = CreateRuntimeInstance(sourceCompilation);
             var context = CreateMethodContext(runtime, methodName);
 
             VerifyHasThis(context, expectedType, expectedIL);
+
+            // Now recompile and test CompileExpression with optimized code.
+            sourceCompilation = sourceCompilation.WithOptions(sourceCompilation.Options.WithOptimizationLevel(OptimizationLevel.Release));
```

Comment:
I think it would be better to pass an additional flag indicating whether ```this``` is expected to be present in release builds.  If so, confirm that it is (with IL, if possible).  Otherwise, compile ```1``` as a sanity check.